### PR TITLE
Sanlouise/refactor addressformatter

### PR DESCRIPTION
**Pull request by sanlouise**
[TICKET](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8596)

### Description
While working on the international addresses preview, I noticed that we have some dead/outdated code in the `addressFormatter` helper function. We should especially clean up how we format military addresses.

We need to use the `addressFormatter` in the `AddressValidationModal` as well to ensure consistency among address formatting in our profile.

### Acceptance Criteria
- [x] Investigate if we are ever using `militaryPostOfficeTypeCode` and `militaryStateCode`
- [x] Undo passing in `type` to the `addressFormatter`
- [x] If first acceptance criteria is `false`, remove `case` statement for `military`, instead bucket military addresses with domestic ones
- [x] Use `addressFormatter` in `src/platform/user/profile/vet360/containers/AddressValidationModal.jsx`
- [x] Ensure checkbox on Address Confirmation modal is not stretched out when using long addresses
- [x] Ensure unit tests are updated

On the address validation modal we now list alle `streetLines` on the same line, and wrap to the nest line when we have long strings:

image

### CODE REVIEW

**Comment by erikphansen** [comment]
Looking at the swagger docs [here](https://department-of-veterans-affairs.github.io/va-digital-services-platform-docs/api-reference/#/user/getUser) and it's clear that the only valid address types are `DOMESTIC, INTERNATIONAL, OVERSEAS MILITARY `. I wonder if `MILITARY` was a holdover from when address data came from something other than the VA Profile service. @ncksllvn does that seem possible? 
image

**Comment by ncksllvn** [comment]
@erikphansen I _think_ that address stuff originates with Letters and an EVSS endpoint. It was shared when VA Profile became a thing, but it ended up creating some wild stuff. So yeah, there are definitely possible remnants hanging around

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/AddressValidationModal.jsx:179]: I think we should stick with wrapping each of the three address elements in `span`s like we were doing before. Also if `country` isn't a string with a length > 0 we'll want to be sure to not render anything for it.

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/AddressValidationModal.jsx:153]: we could just pass the `address` directly without the spread but I know some people prefer this "just in case" the function would mess with the passed-in arg.

**Comment by sanlouise** [comment]
> here

@erikphansen Thanks for sharing this doc and confirming! This is very helpful 👍

**Review by erikphansen — APPROVED** [review]
_I_ don't see anything wrong with deleting this old code. And I really like being able to delete the address formatting code that was being used in the validation modal (I would have told Brad to use the existing address formatter if I knew it existed!). But I really think that Nick should give this a look before we merge it.

**Comment by erikphansen** [comment]
this test is legit failing in Jenkins src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx:400:10

**Comment by sanlouise** [comment]
> this test is legit failing in Jenkins src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx:400:10

Yeah, I will fix that before putting up a PR for review - I think I know what's up 👍 I just wanted to make sure that I was at least going about this cleanup the right way

**Comment by ncksllvn** [comment]
Oh yeah, I think a lot of this was used for normalizing a VA Profile-format address into a Letters-format address before being passed into the Letters Address edit modal, then trasnformed back after the edit is finished! So glad to see the tech debt eliminated

File: src/platform/forms/address/helpers.js
Comment (ncksllvn) [inline src/platform/forms/address/helpers.js:101]: Should this comment say to `only look up for ADDRESS_TYPES.domestic`?

**Review by ncksllvn — APPROVED** [review]
Really great work! Only thing to consider is possibly strengthening the E2E test for the address updates on the Profile - `src/applications/personalization/profile360/tests/e2e/profile-360.e2e.spec.js` - to tie API payloads into the UI and get that extra reassurance. But otherwise this is awesome 🔥 

Aaand my only comment is a comment about a comment 🙂

File: src/platform/user/profile/vet360/components/AddressField/AddressView.jsx
Comment (sanlouise) [reply src/platform/user/profile/vet360/components/AddressField/AddressView.jsx:7]: Good catch!

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx
Comment (sanlouise) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx:153]: I'll make the change to just pass in the address without the spread :)

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx
Comment (sanlouise) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx:179]: I'll make sure to conditionally render the country! 👍 

For the `addressLines` however, we were rendering all 3 properties on one line on the Profile. The only place we were rendering them on spans on separate lines was in the address validation modal. I think we should stick with the same format in both places, and I chose to go with formatting on the Profile - I got approval from Matt to get the address to render on one line on the validation modal as well.

I could render separate spans, but then I'd need to explicitly return all separate lines from the `formatAddress` function, and then I'd also like to change how we render the address on the Profile for consistency. If it's up to me, I'd keep the current implementation 
 (rendering on one line) if you don't have a strong opinion on this :)

File: src/platform/forms/address/helpers.js
Comment (sanlouise) [reply src/platform/forms/address/helpers.js:101]: That's much clearer :smile:

File: src/platform/user/profile/vet360/containers/AddressValidationModal.jsx
Comment (erikphansen) [reply src/platform/user/profile/vet360/containers/AddressValidationModal.jsx:177]: sorry I meant wrap `street` `cityStateZip` and `country` each in their own span. Keeping one `street` is fine. Then you don't have to add the `br` tags (and won't have to worry about conditionally adding the `br` before `country` depending on if `country` is set). The fact that the container is a flex container means that the `span`s will be treated like block elements and stack on either other vertically.

File: src/platform/forms/tests/address.unit.spec.js
Comment (sanlouise) [inline src/platform/forms/tests/address.unit.spec.js:79]: Checking with Peggy, we want to display the full string.

File: src/platform/forms/tests/address.unit.spec.js
Comment (sanlouise) [inline src/platform/forms/tests/address.unit.spec.js:67]: Checked with Peggy, we abbreviate US states.

File: src/applications/personalization/profile360/sass/profile-360.scss
Comment (kelsonic) [inline src/applications/personalization/profile360/sass/profile-360.scss:90]: @erikphansen or @ncksllvn, do you know if there's a variable we could use here instead of the `1.4rem`?

Comment (sanlouise) [reply src/applications/personalization/profile360/sass/profile-360.scss:90]: For context - I needed to add this to make sure the checkbox does not appear stretched when `{street}` is such a long string that it covers more than a single line.

File: src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx
Comment (kelsonic) [inline src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx:402]: Did we mean to not have a space between `23` and `Tampa`?

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

File: src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx
Comment (sanlouise) [reply src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx:402]: Tampa is rendered on a new line 😄

File: src/applications/personalization/profile360/sass/profile-360.scss
Comment (erikphansen) [reply src/applications/personalization/profile360/sass/profile-360.scss:90]: Not really, @kelsonic. Looks like doing `units(1.5)` would result in 1.2rem and `units(2)` would result in 1.6rem https://design.va.gov/design/spacing-units

File: src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx
Comment (erikphansen) [reply src/platform/user/profile/vet360/tests/containers/AddressValidationModal.unit.spec.jsx:402]: No big deal but I'm curious if there's a way to check for `\n` or something?

**Review by erikphansen — APPROVED** [review]

**Comment by erikphansen** [comment]
Oh no! looks like a legit e2e test failure:
```
TEST FAILURE:  1 assertions failed, 1427 passed. (7m 19s)

  ✖ applications/personalization/profile360/tests/e2e/profile-360.e2e.spec
    - Begin application (6.913s)
    Testing if element <[data-field-name="mailingAddress"]> contains text: "Fulton, New York 97062".  - expected "Fulton, New York 97062" but got: "Mailing address Edit
    1493 Martin Luther King Rd, Apt 1
    Fulton, NY 97062"
        at containsText (/application/src/applications/personalization/profile360/tests/e2e/profile-360.e2e.spec.js:40:18)
        at runAddressTest (/application/src/applications/personalization/profile360/tests/e2e/profile-360.e2e.spec.js:67:3)
        at beginTests (/application/src/applications/personalization/profile360/tests/e2e/profile-360.e2e.spec.js:110:7)
        at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by sanlouise** [comment]
> personalization/profile360/tests/e2e/profile-360.e2e.spec

Oops, fixed! We now only want to render the abbreviated version for US states.

**Comment by erikphansen** [comment]
Looks good now!

**Comment by sanlouise** [comment]
@department-of-veterans-affairs/frontend-review-group this PR is ready for review whenever you have a moment, thank you!

**Review by rianfowler — APPROVED** [review]

**Pull request merged**
